front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -235,25 +235,27 @@ export async function generateRetrievalSpecification(
 // Internal interface for the retrieval and rendering of a Retrieval action. This should not be used
 // outside of api/assistant. We allow a ModelId interface here because we don't have `sId` on
 // actions (the `sId` is on the `Message` object linked to the `UserMessage` parent of this action).
-export async function renderRetrievalActionsByModelId(
-  ids: ModelId[]
+export async function retrievalActionTypesFromAgentMessageIds(
+  agentMessageIds: ModelId[]
 ): Promise<RetrievalActionType[]> {
-  const actionById = (
-    await AgentRetrievalAction.findAll({
-      where: {
-        id: ids,
-      },
-    })
-  ).reduce<{
+  const models = await AgentRetrievalAction.findAll({
+    where: {
+      agentMessageId: agentMessageIds,
+    },
+  });
+
+  const actionById = models.reduce<{
     [id: ModelId]: AgentRetrievalAction;
   }>((acc, a) => {
     acc[a.id] = a;
     return acc;
   }, {});
 
-  if (Object.keys(actionById).length !== ids.length) {
+  const ids = models.map((a) => a.id);
+
+  if (Object.keys(actionById).length !== models.length) {
```

Comment:
that doesn't work, you're no longer testing that `models` has indeed all the models from `agentMessageIds` no ?

Reply:
Also I guess this can never be true, since SQL Where IN() will never return duplicates even if the array has duplicates

Reply:
@spolu you decided to not fix this ?

Reply:
Yep the idea is that if we have duplicate Ids (as input) we would error here because the returned models would not have duplicates?

Reply:
There is no possibility for your condition to be false ever

Reply:
Thx: will fix in subsequent PR